### PR TITLE
Adds conditional reveal to "another website" checkbox

### PR DESCRIPTION
**Pull request by davidtrussler**
[Trello card](https://trello.com/c/OoA1ipPB/773-bug-missing-conditional-external-url-input-on-held-on-another-website-field)

This change restores a conditional input on the "Edit consultation" page which permanently hidden after porting to the Design System. This led to the user receiving an error message ("External URL can't be blank") if they checked the checkbox but they had no option to provide a value for the URL.

| Before | After |
| --- | --- |
| Screenshot 2022-09-27 at 15 45 43 | Screenshot 2022-09-27 at 15 44 36 |

### CODE REVIEW

File: app/views/admin/consultations/_form.html.erb [hunk 1]
```diff
@@ -6,36 +6,37 @@
 
 <%= standard_edition_form(edition, @information, preview_design_system: true) do |form| %>
   <%= render "govuk_publishing_components/components/fieldset", {
-    legend_text: "Held on another website",
-    heading_level: 3,
-    heading_size: "l"
+    legend_text: "",
```
Comment (DilwoarH) [inline app/views/admin/consultations/_form.html.erb:9]: Is this needed?

File: app/views/admin/consultations/_form.html.erb [hunk 2]
```diff
@@ -6,36 +6,37 @@
...
   } do %>
+    <%= form.hidden_field :external, value: "0" %>
+
     <%= render "govuk_publishing_components/components/checkboxes", {
       name: "edition[external]",
       id: "edition_external",
+      heading: "Held on another website",
+      heading_level: 3,
+      heading_size: "l",
+      hint_text: "",
```
Comment (DilwoarH) [inline app/views/admin/consultations/_form.html.erb:19]:
```suggestion
```

File: app/views/admin/consultations/_form.html.erb [hunk 3]
```diff
@@ -6,36 +6,35 @@
...
 <%= standard_edition_form(edition, @information, preview_design_system: true) do |form| %>
   <%= render "govuk_publishing_components/components/fieldset", {
-    legend_text: "Held on another website",
-    heading_level: 3,
-    heading_size: "l"
+    legend_text: "",
   } do %>
     <%= render "govuk_publishing_components/components/checkboxes", {
       name: "edition[external]",
       id: "edition_external",
```
Comment (DilwoarH) [inline app/views/admin/consultations/_form.html.erb:15]: just drop this ID as we don't need it for anything so the JS won't have any effect

File: app/views/admin/consultations/_form.html.erb (hunk 1 above)
Comment (davidtrussler) [reply app/views/admin/consultations/_form.html.erb:9]: If `legend_text` is left out completely it fires an error in the component even though the `<legend>` is populated by the `heading` variable from the checkbox component. So it looks like we need it though it looks a bit odd. 

```
ActionView::Template::Error (undefined local variable or method `legend_text' for #<ActionView::Base:0x000000004e66c8>
Did you mean?  legend_classes):
    35:         %>
    36:       <% end %>
    37:     <% else %>
    38:       <%= tag.legend legend_text, class: legend_classes %>
    39:     <% end %>
    40:     <% if error_message %>
    41:       <%= render "govuk_publishing_components/components/error_message", {
```

File: app/views/admin/consultations/_form.html.erb (hunk 2 above)
Comment (davidtrussler) [reply app/views/admin/consultations/_form.html.erb:19]: 👍 Yes, we can do without that.

File: app/views/admin/consultations/_form.html.erb (hunk 3 above)
Comment (davidtrussler) [reply app/views/admin/consultations/_form.html.erb:15]: 👍 Good point - removed.

**Review by davidgisbey — APPROVED** [review]
lgtm

**Pull request merged**